ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/BundleManifestSourceLocationManager.java

```diff
@@ -45,7 +51,9 @@ public class BundleManifestSourceLocationManager {
 	/**
 	 * Maps SourceLocationKeys (plugin name and version) to IPluginModelBase objects representing source bundles
 	 */
-	private Map<SourceLocationKey, IPluginModelBase> fPluginToSourceBundle = new LinkedHashMap<>(0);
+	private Map<SourceLocationKey, IPluginModelBase> fPluginToSourceBundle = Map.of();
+
+	private Map<SourceLocationKey, TargetBundle> fPluginToTargetBundle = Map.of();
```

Comment:
One thing that could be improved to map SourceLocationKeys directly to the SourceLocation. That would save the computation in each look-up and would probably simplify `getSourceLocations()`.

But I'm fine with doing that later. :)

Reply:
I alligned this with how pde aready works, and even if it should be not required, it seems this is for cases where the location might has changed but I can't find a reall use-case for that.

Reply:
Yes, that would be different and then should also be changed for the existing code.
It was more a general remark for a potential improvement that could be done later.

I would assume that the location can change at all without re-loading the target?